Test failures are ignored when filename is null in the stack trace

Example test code (based on scala-js/scalajs-tutorial):
```
package tutorial.webapp

import utest._
import scala.scalajs.js

@js.native
trait Foo extends js.Object {
 var x: js.Dictionary[Int] = js.native
}

object TutorialTest extends TestSuite {

  def tests = Tests {
    'HelloWorld - {
      val foo = js.Dynamic.literal(x = null).asInstanceOf[Foo]
      assert(foo.x.values.exists(_ > 0))
    }
  }
}
```

and the result:

```
sbt:Scala.js Tutorial> test
-------------------------------- Running Tests --------------------------------
X tutorial.webapp.TutorialTest.HelloWorld 4ms
  utest.AssertionError: assert(foo.x.values.exists(_ > 0))
  foo: tutorial.webapp.Foo = [object Object]
  scala.scalajs.js.JavaScriptException: TypeError: Cannot convert undefined or null to object
    <jscode>.Function.keys@<anonymous>(Unknown)
    scala.scalajs.js.WrappedDictionary$DictionaryIterator.<init>(R:\scalajs-tutorial-0.6.x\target\sc
    ala-2.12\scala-js-tutorial-test-fastopt.js:23811)
    scala.scalajs.js.WrappedDictionary.iterator(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js
    -tutorial-test-fastopt.js:46729)
    scala.collection.MapLike$$anon$2.<init>(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tut
    orial-test-fastopt.js:26890)
    <jscode>.$c_sjs_js_WrappedDictionary.$c_sc_AbstractMap.valuesIterator__sc_Iterator(R:\scalajs-tu
    torial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:40509)
    <jscode>.{anonymous}()(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastop
    t.js:17605)
    scala.scalajs.runtime.AnonFunction1.apply(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-t
    utorial-test-fastopt.js:20908)
    utest.asserts.Util$.runAssertionEntry(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutor
    ial-test-fastopt.js:6632)
[error] scala.scalajs.js.JavaScriptException: TypeError: Cannot read property 'length' of null
[error]         at java.io.DataOutputStream.writeUTF(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:31125)
[error]         at org.scalajs.testcommon.Serializer$StackTraceElementSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13354)
[error]         at org.scalajs.testcommon.Serializer$StackTraceElementSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13346)
[error]         at org.scalajs.testcommon.Serializer$$anon$4.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:12834)
[error]         at org.scalajs.testcommon.Serializer$ThrowableSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13570)
[error]         at org.scalajs.testcommon.Serializer$ThrowableSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13577)
[error]         at org.scalajs.testcommon.Serializer$$anon$5.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:12887)
[error]         at org.scalajs.testcommon.Serializer$ThrowableSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13574)
[error]         at org.scalajs.testcommon.Serializer$OptionalThrowableSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13199)
[error]         at org.scalajs.testcommon.Serializer$EventSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:12985)
    <jscode>.{anonymous}()(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastop
    t.js:17635)
    scala.scalajs.runtime.AnonFunction0.apply(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-t
    utorial-test-fastopt.js:20880)
[info]
[success] Total time: 5 s, completed 2019/04/08 20:58:04
```

Example 2:

```
package tutorial.webapp

import utest._
import scala.scalajs.js

object TutorialTest extends TestSuite {

  def tests = Tests {
    'HelloWorld - {
      js.Dynamic.global.eval("throw new Error()")
      assert(true)
    }
  }
}
```

and the result:

```
...
-------------------------------- Running Tests --------------------------------
X tutorial.webapp.TutorialTest.HelloWorld 1ms
  scala.scalajs.js.JavaScriptException: Error
    <jscode>.eval(eval at <anonymous> (R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial
    -test-fastopt.js:17595:21), <anonymous>:1)
    <jscode>.eval@<anonymous>(Unknown)
    <jscode>.{anonymous}()(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastop
    t.js:17595)
    scala.scalajs.runtime.AnonFunction0.apply(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-t
    utorial-test-fastopt.js:20868)
[error] scala.scalajs.js.JavaScriptException: TypeError: Cannot read property 'length' of null
[error]         at java.io.DataOutputStream.writeUTF(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:30907)
...
```

Although utest clearly prints about test failure, `test` task was finished as `[success]` and  TestsFailedException hasn't be thrown.

I think this JavaScriptException has a StackTraceElement with filename=null, and the filename (null) has been passed to DataOutputStream.writeUTF which doesn't expect null.

sbt 1.2.8 / scala 2.12.6 / sbt-scalajs 0.6.25 / utest 0.6.3

--- a/test-common/src/main/scala/org/scalajs/testcommon/Serializer.scala
+++ b/test-common/src/main/scala/org/scalajs/testcommon/Serializer.scala
@@ -132,13 +132,13 @@ private[scalajs] object Serializer {
     def serialize(x: StackTraceElement, out: SerializeState): Unit = {
       out.write(x.getClassName())
       out.write(x.getMethodName())
-      out.write(x.getFileName())
+      out.write(Option(x.getFileName()).getOrElse(""))
       out.write(x.getLineNumber())
     }
 
     def deserialize(in: DeserializeState): StackTraceElement = {
       new StackTraceElement(in.read[String](), in.read[String](),
-          in.read[String](), in.read[Int]())
+          Option(in.read[String]()).filter(_.nonEmpty).orNull, in.read[Int]())
     }
   }
 

--- a/test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala
+++ b/test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala
@@ -28,4 +28,13 @@ class SerializerTest {
     assertEquals(in.toString(), out.toString())
     assertEquals(in.getStackTrace().size, out.getStackTrace().size)
   }
+
+  // # 3611
+  @Test
+  def serializeStackTraceElementWithNullFilename: Unit = {
+    val st = new StackTraceElement("MyClass", "myMethod", null, 1)
+    val deserialized = roundTrip(st)
+    assertNull(deserialized.getFileName)
+    assertEquals("MyClass.myMethod(Unknown Source)", deserialized.toString)
+  }
 }

